spec/authorization_spec.rb

```diff
@@ -271,4 +271,38 @@ def to_params(*args, **kwargs, &block)
       expect(Controller.new(user, action, params).permitted_attributes(post, :revise).to_h).to eq("body" => "blah")
     end
   end
+
+  describe "#pundit_reset!" do
+    let(:new_user) { double }
+
+    it "clears the current user" do
+      expect(controller.pundit.user).to eq controller.current_user
+
+      controller.current_user = new_user
+      expect(controller.pundit.user).not_to eq controller.current_user
+
+      controller.pundit_reset!
+      expect(controller.pundit.user).to eq controller.current_user
+    end
+
+    it "clears the policy cache" do
+      controller.authorize(post)
+      expect(controller.policies).not_to be_empty
+      expect(controller.pundit_policy_authorized?).to be true
```

Comment:
I understand how this assertion ended up in this test, but I believe it's out of place.

`pundit_policy_authorized?` (and `pundit_policy_scoped?`) doesn't have anything to do with the cache, it's a flag set by `authorize` to help catch errors in development: https://github.com/varvet/pundit?tab=readme-ov-file#ensuring-policies-and-scopes-are-used

As such, I believe these two methods should have their own tests using `#pundit_reset!`.